src/psyclone/psyir/transformations/inline_trans.py

```diff
@@ -0,0 +1,478 @@
+                    f"Return statements and therefore cannot be inlined.")
+
+        if routine.walk(CodeBlock):
+            raise TransformationError(
+                f"Routine '{name}' contains one or more "
+                f"CodeBlocks and therefore cannot be inlined.")
+
+        # Support for routines with named arguments is not yet implemented.
+        # TODO #924.
+        for arg in node.argument_names:
+            if arg:
+                raise TransformationError(
+                    f"Routine '{routine.name}' cannot be inlined because it "
+                    f"has a named argument '{arg}' (TODO #924).")
+
+        # Check for symbol-naming clashes that we can't handle.
+        table = node.scope.symbol_table
+        routine_table = routine.symbol_table
+
+        # We can't handle a clash between (apparently) different symbols that
+        # share a name but are imported from different containers.
+        callsite_imports = table.imported_symbols
+        routine_imports = routine_table.imported_symbols
+        routine_import_names = [sym.name for sym in routine_imports]
+        for sym in callsite_imports:
+            if sym.name in routine_import_names:
+                routine_sym = routine_table.lookup(sym.name)
+                if (routine_sym.interface.container_symbol.name !=
+                        sym.interface.container_symbol.name):
+                    raise TransformationError(
+                        f"Routine '{routine.name}' imports '{sym.name}' from "
+                        f"Container "
+                        f"'{routine_sym.interface.container_symbol.name}' but "
+                        f"the call site has an import of a symbol with the "
+                        f"same name from Container "
+                        f"'{sym.interface.container_symbol.name}'.")
+
+        # Check for unresolved symbols or for any accessed from the Container
+        # containing the target routine.
+        refs = routine.walk(Reference)
+        for ref in refs:
+            if ref.symbol.name not in routine_table:
+                sym = routine_table.lookup(ref.symbol.name)
+                if sym.is_unresolved:
+                    raise TransformationError(
+                        f"Routine '{routine.name}' cannot be inlined because "
+                        f"it accesses an un-resolved variable "
+                        f"'{ref.symbol.name}'.")
+                if not sym.is_import:
+                    raise TransformationError(
+                        f"Routine '{routine.name}' cannot be inlined because "
+                        f"it accesses variable '{ref.symbol.name}' from its "
+                        f"parent container.")
+
+        # Check that the shape of any dummy array arguments are the same as
+        # those at the call site.
+        visitor = FortranWriter()
+        for dummy_arg, actual_arg in zip(routine_table.argument_list,
```

Comment:
Maybe you can add a `# pylint: disable=cell-var-from-loop` here explaining that the the loop variables we are capturing are fine because the loop is ended after capturing them.

Reply:
Oh yes, I had to google that!